File: server/module/module-erp/erp-server/src/convert/erp_sales_return_detail.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_sales_return_detail::{self, Model as ErpSalesReturnDetail, ActiveModel as ErpSalesReturnDetailActiveModel};
use erp_model::request::erp_sales_return_detail::{CreateErpSalesReturnDetailRequest, UpdateErpSalesReturnDetailRequest};
use erp_model::response::erp_sales_return_detail::{ErpSalesReturnDetailBaseResponse, ErpSalesReturnDetailResponse};

pub fn create_request_to_model(request: &CreateErpSalesReturnDetailRequest) -> ErpSalesReturnDetailActiveModel {
    ErpSalesReturnDetailActiveModel {
        sale_detail_id: Set(request.sale_detail_id.clone()),
        warehouse_id: Set(request.warehouse_id.clone()),
        quantity: Set(request.quantity.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpSalesReturnDetailRequest, existing: ErpSalesReturnDetail) -> ErpSalesReturnDetailActiveModel {
    let mut active_model: ErpSalesReturnDetailActiveModel = existing.into();
    if let sale_detail_id = &request.sale_detail_id { 
        active_model.sale_detail_id = Set(sale_detail_id.clone());
    }
    if let warehouse_id = &request.warehouse_id { 
        active_model.warehouse_id = Set(warehouse_id.clone());
    }
    if let quantity = &request.quantity { 
        active_model.quantity = Set(quantity.clone());
    }
    if let Some(remarks) = &request.remarks { 
        active_model.remarks = Set(Some(remarks.clone()));
    }
    active_model
}

pub fn model_to_response(model: ErpSalesReturnDetail) -> ErpSalesReturnDetailResponse {
    ErpSalesReturnDetailResponse { 
        id: model.id,
        order_id: model.order_id,
        sale_detail_id: model.sale_detail_id,
        warehouse_id: model.warehouse_id,
        product_id: model.product_id,
        quantity: model.quantity,
        unit_price: model.unit_price,
        subtotal: model.subtotal,
        tax_rate: model.tax_rate,
        remarks: model.remarks,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

pub fn model_to_base_response(model: ErpSalesReturnDetail) -> ErpSalesReturnDetailBaseResponse {
    ErpSalesReturnDetailBaseResponse { 
        id: model.id,
        sale_detail_id: model.sale_detail_id,
        warehouse_id: model.warehouse_id,
        remarks: model.remarks,
    }
}

File: admin/src/api/erp_sales_return_detail.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/erp/erp_sales_return_detail/create', // 新增
  update: '/erp/erp_sales_return_detail/update', // 修改
  delete: '/erp/erp_sales_return_detail/delete', // 删除
  get: '/erp/erp_sales_return_detail/get', // 单条查询
  list: '/erp/erp_sales_return_detail/list', // 列表查询
  page: '/erp/erp_sales_return_detail/page', // 分页查询
}

export interface ErpSalesReturnDetailRequest {
  id: number; // 退货详情ID
  sale_detail_id: number; // 销售订单详情ID
  warehouse_id: number; // 仓库ID
  product_id?: number; // 产品ID
  quantity: number; // 数量
  unit_price?: number; // 单价
  subtotal?: number; // 小计
  tax_rate?: number; // 税率,精确到万分位
  remarks?: string; // 备注
}

export interface ErpSalesReturnDetailResponse {
  id: number; // 退货详情ID
  order_id: number; // 退货订单ID
  sale_detail_id: number; // 销售订单详情ID
  warehouse_id: number; // 仓库ID
  product_id: number; // 产品ID
  quantity: number; // 数量
  unit_price: number; // 单价
  subtotal: number; // 小计
  tax_rate: number; // 税率,精确到万分位
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpSalesReturnDetailBaseResponse {
  id: number; // 退货详情ID
  sale_detail_id: number; // 销售订单详情ID
  warehouse_id: number; // 仓库ID
  product_id: number; // 产品ID
  quantity: number; // 数量
  unit_price: number; // 单价
  subtotal: number; // 小计
  tax_rate: number; // 税率,精确到万分位
  remarks: string; // 备注
}

export interface ErpSalesReturnDetailQueryCondition extends PaginatedRequest {

}

export const createErpSalesReturnDetail = (erp_sales_return_detail: ErpSalesReturnDetailRequest): Promise<number> => {
  return api.post<number>(apis.create, erp_sales_return_detail);
}

export const updateErpSalesReturnDetail = (erp_sales_return_detail: ErpSalesReturnDetailRequest): Promise<void> => {
  return api.post<void>(apis.update, erp_sales_return_detail);
}

export const deleteErpSalesReturnDetail = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getErpSalesReturnDetail = (id: number): Promise<ErpSalesReturnDetailResponse> => {
  return api.get<ErpSalesReturnDetailResponse>(`${apis.get}/${id}`);
}

export const listErpSalesReturnDetail = (): Promise<Array<ErpSalesReturnDetailResponse>> => {
  return api.get<Array<ErpSalesReturnDetailResponse>>(apis.list);
}

export const pageErpSalesReturnDetail = (condition: ErpSalesReturnDetailQueryCondition): Promise<PaginatedResponse<ErpSalesReturnDetailResponse>> => {
  return api.get<PaginatedResponse<ErpSalesReturnDetailResponse>>(apis.page, condition);
}
